### DISCUSSION TITLE
How to use shared surfingkeys config file with Safari extension?

### DISCUSSION THREAD

**Discussion by megalithic** [brookhong/Surfingkeys#1790] [Q&A]
I just purchased surfingkeys extension for safari and am not seeing how I can use my local surfingkeys config (that i presently use across chrome/brave/firefox -- though firefox uses my github repo url). Please advise. Thanks!

**Comment by megalithic** [comment]
No one using surfingkeys + safari?

**Reply by megalithic** [reply]
ping @brookhong

**Comment by b0o** [comment]
You can do this in a hacky way. Open the SurfingKeys settings page, then open the devtools console, and run `document.getElementById("localPathForSettings").style.display = ""`. This should unhide the "Load settings from:" input and allow you to type in a URL and click save.

**Comment by brookhong** [comment]
There are some limitations from Safari itself, reading content from URL in Javascript is not allowed, that why the input element on the setting page is also hidden.

Sorry for that, I should also mention this on https://github.com/brookhong/Surfingkeys#feature-availability, will update it soon.

**Reply by b0o** [reply]
Weird because reading from URL works for me in Safari on macOS.

**Reply by brookhong** [reply]
Reading from a HTTP URL works, but reading from a file:/// URL is not allowed. Maybe we can turn the feature on for Safari but with a different description for the input box.

**Reply by b0o** [reply]
That would be nice. Do you know if reading from an HTTP URL works in Safari on iPadOS/iOS? I can't easily do the workaround I mentioned there since I can't use the devtools.
